docs/spec/staking/02_state_transitions.md

```diff
@@ -2,96 +2,114 @@
- - remove the validator if it is unbonded and there are no more delegation shares. 
+Delegation may be called.
+
+- subtract the unbonded shares from delegator
+- update the delegation or remove the delegation if there are no more shares
+- if the delegation is the operator of the validator and no more shares exist then trigger a jail validator
+- update the validator with removed the delegator shares and associated coins
+- if the validator state is `Bondedd`, transfer the `Coins` worth of the unbonded
+shares from the `BondedPool` to the `NotBondedTokens` `ModuleAccount`
+- remove the validator if it is unbonded and there are no more delegation shares.
 
 ### Undelegate
-When an delegation occurs both the validator and the delegtion objects are affected  
- - perform an unbond delegation 
-   - if the validator is unbonding or bonded add the tokens to an
-     `UnbondingDelegation` Entry
-   - if the validator is unbonded send the tokens directly to the withdraw
-     account
+
+When an delegation occurs both the validator and the delegation objects are affected  
+
+- perform an unbond delegation
+- if the validator is `Unbonding` or `Bonded` add the tokens to an `UnbondingDelegation` Entry
+- if the validator is `Unbonded` send the tokens directly to the withdraw
+  account
 
 ### Complete Unbonding
+
 For undelegations which do not complete immediately, the following operations
 occur when the unbonding delegation queue element matures:
- - remove the entry from the `UnbondingDelegation` object
- - withdraw the tokens to the delegator withdraw address 
+
+- remove the entry from the `UnbondingDelegation` object
+- transfer the tokens from the `NotBondedTokens` `ModuleAccount` to the delegator `Account`
 
 ### Begin Redelegation
-Redelegations affect the delegation, source and destination validators. 
- - perform an unbond delegation from the source validator 
- - using the generated tokens perform a Delegate to the destination
-   validator
- - record the token amount in an new entry in the relevant `Redelegation`
+
+Redelegations affect the delegation, source and destination validators.
+
+- perform an `unbond` delegation from the source validator to retrieve the tokens
+worth of the unbonded shares
+- using the unbonded tokens, `Delegate` them to the destination validator
+- if the `sourceValidator.Status` is `Bonded`, and the `destinationValidator` is not, transfer the newly delegated tokens to the `NotBondedPool` `ModuleAccount`
+- otherwise, if the `sourceValidator.Status` is not `Bonded`, and the `destinationValidator` is `Bonded`, transfer the newly delegated tokens to the `BondedPool` `ModuleAccount`
+- record the token amount in an new entry in the relevant `Redelegation`
 
 ### Complete Redelegation
+
 When a redelegations complete the following occurs:
- - remove the entry from the `Redelegation` object
 
+- remove the entry from the `Redelegation` object
 
-TODO TODO TOFU TODO
```

Comment:
lol

Reply:
the `Slashing` section is incomplete as well